src/jpgimage.cpp

```diff
@@ -641,7 +641,7 @@ namespace Exiv2 {
                 // Read size and signature
                 std::memset(buf.pData_, 0x0, buf.size_);
                 bufRead = io_->read(buf.pData_, bufMinSize);
-                if (io_->error())
+                if (io_->error() || bufRead != bufMinSize)
                     throw Error(kerFailedToReadImageData);
                 if (bufRead < 2)
```

Comment:
I think this `if()` is now obsolete because you are already checking `bufRead != 36` in the if above.

Reply:
You're right. I have removed it now. Thanks!